File: app/src/main/java/com/capstone/project_niyakneyak/ui/main/fragment/viewmodel/CheckListViewModelFactory.java
package com.capstone.project_niyakneyak.ui.main.fragment.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class CheckListViewModelFactory extends ViewModelProvider.AndroidViewModelFactory {
    private Application application;
    public CheckListViewModelFactory(Application application){this.application = application;}
    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(CheckListViewModel.class)) {
            return (T) new CheckListViewModel(application);
        } else {
            throw new IllegalArgumentException("Unknown ViewModel class");
        }
    }
}

File: app/src/main/java/com/capstone/project_niyakneyak/ui/main/fragment/viewmodel/CheckListViewModelFactory.kt
package com.capstone.project_niyakneyak.ui.main.fragment.viewmodel

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class CheckListViewModelFactory(private val application: Application) : ViewModelProvider.AndroidViewModelFactory() {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return if (modelClass.isAssignableFrom(CheckListViewModel::class.java)) {
            CheckListViewModel(application) as T
        } else {
            throw IllegalArgumentException("Unknown ViewModel class")
        }
    }
}